drivers/w1/CMakeLists.txt

```diff
@@ -8,6 +8,11 @@ zephyr_library_sources(w1_common.c)
 # drivers implementing link functions (read, write, reset)
 zephyr_library_sources_ifdef(CONFIG_W1_TEST           w1_test.c)
 zephyr_library_sources_ifdef(CONFIG_W1_ZEPHYR_SERIAL  w1_zephyr_serial.c)
+zephyr_library_sources_ifdef(CONFIG_W1_DS2484         w1_ds2484.c)
+
+if(CONFIG_W1_DS2484)
+	zephyr_library_sources(w1_ds248x.c)
```

Comment:
why isn't this just `zephyr_library_sources_ifdef(CONFIG_W1_DS2484 w1_ds248x.c)`

Reply:
Precisely. It should already be in this PR but I found some incompatibilities and didn't want to withdraw the PR altogether. The other drivers will follow.